### ISSUE TITLE
Add a gesture for repeating an instruction

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#979]
Some navigation applications repeat the last spoken instruction when you tap on the turn banner. This way, you can slam your thumb on the top part of the screen without taking your eyes off the road. Nowadays, tapping our turn banner brings down the step table, so we’d have to choose a different gesture for repeating the instruction. Long-pressing, maybe? 🤔

/cc @mapbox/navigation-ios

**Comment by bsudekum** [comment]
Another idea I'd like to throw into the mix- after unmuting, and it has been a sufficient amount of time since the previous announcement, repeat the current instruction. Ideally, rapid pressing the mute switch would not announce the instruction.

**Comment by bsudekum** [comment]
@1ec5 @vincethecoder how about as a first pass, when tapping the user puck, the instruction is repeated?

**Comment by JThramer** [comment]
Personally I would suggest tapping the instructions banner to repeat an instruction. This is similar functionality to other navigation applications.

Of course, right now tapping the banner shows the steps table. Perhaps we could switch that to a pan-down gesture?

My 2¢ anyways.

**Comment by bsudekum** [comment]
Yeah this is tricky, a pan down gesture is sort of hidden. Long term, toggling the steps banner view would enabled via a button.

**Comment by 1ec5** [comment]
> how about as a first pass, when tapping the user puck, the instruction is repeated?

I don’t think that would be discoverable. No one associates the user puck with instructions of any sort.

> Personally I would suggest tapping the instructions banner to repeat an instruction. This is similar functionality to other navigation applications.

Count me in with the tap-to-repeat crowd. We’ve observed in user studies that people have a difficult time discovering the step table when it’s hidden behind a tap of the banner. People keep trying to swipe up from the bottom bar, which is where we used to put it, or down from the banner. Nothing in the banner indicates that it hides a full list – not even a 🍔 icon.

If revisiting that design is infeasible for some reason, a half-solution would be to repeat the instruction when long-pressing or force-tapping the banner. Note that, in some jurisdictions, it’s unlawful to perform any gesture more complex than a swipe or tap while driving.

**Comment by 1ec5** [comment]
#1150 and #1253 made it easier to access the steps table, but neither PR actually added a tap-to-repeat affordance, as requested in this issue.

To avoid misleading the user, we need to regenerate a new instruction each time with the correct distance to the maneuver, rather than reusing the SSML and audio from the previously played instruction. Ideally, the Directions API would provide SSML with a placeholder for the distance, to be filled in based on the current distance to the maneuver. Short of that, I suppose we could build our own format string based on the step’s final voice instruction and the phrases in [OSRM Text Instructions](https://github.com/Project-OSRM/osrm-text-instructions.swift/).
